site/migration/01-migration.md

```diff
@@ -0,0 +1,257 @@
+  - [Kptfile schema changes](#Kptfile-schema-changes)
+  - [`kpt pkg sync` deprecated](#kpt-pkg-sync-deprecated)
+  - [Changes to Functions](#Changes-to-Functions)
+    - [`kpt fn run` deprecated](#kpt-fn-run-deprecated)
+    - [Function Config](#Function-Config)
+    - [Function Results](#Function-Results)
+  - [Changes to Setters](#Changes-to-Setters)
+    - [Setter validations deprecated](#Setter-validations-deprecated)
+    - [Auto-setters deprecated](#Auto-setters-deprecated)
+    - [Function Results](#Function-Results)
+  - [Changes to Live](#changes-to-live)
+- [Migration steps](#Migration-steps)
+  - [Automated portion of migration](#Automated-portion-of-migration)
+    - [Changes made by the function](#Changes-made-by-the-function)
+  - [Manual portion of migration](#Manual-portion-of-migration)
+- [Timeline](#Timeline)
+
+kpt `v1.0` is going to be the latest major release of the kpt CLI. The implementation of kpt `v1.0`
+has changed considerably from kpt `v0.39.0`. A rich set of new features have been added
+as a result of the customers’ input and requirements. Some features have been
+deprecated or refactored in ways that make them incompatible with kpt `v0.39.0`.
+Since these are backwards incompatible changes, there should be a way for customers
+to migrate/fix their existing kpt packages which are compatible with `v0.39.0` version of kpt,
+to become compatible with kpt `v1.0`. This document outlines the end to end migration
+journey of customers using a comprehensive kpt package example.
+
+## What's new and changed?
+
+### CLI changes
+
+To start with, all the commands in kpt `v1.0` will follow the consistent pattern
+
+```shell
+$ kpt <group> <command> <positional_args> [PKG_PATH | DIR | - STDIN] [flags]
+```
+
+Almost all the existing commands/features in kpt `v0.39.0` are also offered in kpt `v1.0`
+but in a better and enhanced format. In kpt `v0.39.0`, `cfg` group had been a dumping ground
+for many cli commands which don’t have a coherent story. `cfg` is often confused with
+`pkg` and vice-versa. As a major step, we removed the `cfg` subgroup and rearranged
+the functionality. Here are the one-liners for each command group in kpt `v1.0`.
+
+1. `pkg:` Commands for composing and describing packages.
+2. `fn:` Commands for running functions for validation and customization.
+3. `live:` Commands for interacting with a live cluster.
+
+`PKG_PATH vs DIR:` Most of the commands accept only `PKG_PATH` as input which means
+the input directory must have a valid Kptfile. Few commands can work on just the
+simple `DIR` with resources. Commands operate on the current working directory by default.
+
+| [Existing Commands]                                                                              | [New Commands]                                                                                                                                                                                                                                   |
```

Comment:
I'd just call this Alpha1 Commands

Reply:
Used version numbers instead as we are using it all across the docs. Users might not have observed v1alpha1 in Kptfile at all.